packages/api-server/api_server/routes/tasks/tasks.py

```diff
@@ -221,7 +222,8 @@ async def post_cancel_task(
 )
 async def post_dispatch_task(
     request: mdl.DispatchTaskRequest = Body(...),
-    task_repo: TaskRepository = Depends(task_repo_dep),
+    task_repo: TaskRepository = Depends(TaskRepository),
+    logger: LoggerAdapter = Depends(get_logger),
```

Comment:
I get an error with the logger, when it is called by https://github.com/open-rmf/rmf-web/pull/931/files#diff-451435e6d653e0de8f927158d4352253989e630ee0b531cdda80f5a1cf060962R52, the `run` function within the `schedule_task` route

```
msg="Task exception was never retrieved
future: <Task finished name='Task-2584' coro=<schedule_task.<locals>.run() done, defined at /usr/local/google/home/aaronchong/workspaces/rmf/logfmt-web/packages/api-server/api_server/routes/tasks/scheduled_tasks.py:50> exception=AttributeError(\"'Depends' object has no attribute 'info'\")>" level="ERROR" ts="2024-04-17T13:24:00.78+0800" src="runners.py:44" funcName="run" exc_info="Traceback (most recent call last):
  File "/usr/local/google/home/aaronchong/workspaces/rmf/logfmt-web/packages/api-server/api_server/routes/tasks/scheduled_tasks.py", line 52, in run
    await post_dispatch_task(req, task_repo)
  File "/usr/local/google/home/aaronchong/workspaces/rmf/logfmt-web/packages/api-server/api_server/routes/tasks/tasks.py", line 273, in post_dispatch_task
    logger.info(resp)
AttributeError: 'Depends' object has no attribute 'info'"
```

Do we need to pass in the logger in that call?

Reply:
Yeah, this was not caught in the type check because it is an optional param. Newer fastapi supports a better way to declare depends which can avoid this in the future.